### ISSUE TITLE
Add possibility to include self-defined Jsonrpsee Client in Wrapper struct

### ISSUE THREAD

**Issue by haerdib** [scs/substrate-api-client#714]
E.g. add new function to create a `JsonrpseeClient` wrapper but with a user defined client here: https://github.com/scs/substrate-api-client/blob/3098812af70d80bdd63f82639f76e5c3e2c36312/src/rpc/jsonrpsee_client/mod.rs#L35-L50
Example:
```rust
pub async fn with_client(client: Arc<Client>) -> Self {
	Self { inner: client }
}
```
I'd also propose to make the inner client public - then the user may change settings any time.

**Comment by masapr** [comment]
I don't think we should make the inner client public. We don't want anyone to change the client during the running program (what happens e. g. with subscriptions, when I exchange the client? What happens with running processes? We would have to handle all these cases.).

But adding a constructor which takes a custom JsonrpseeClient is fine. I don't see any problem with that.

**Comment by haerdib** [comment]
That is true, but this way no one can access the inner convenience functions, such as `is_connected` (see here: https://github.com/paritytech/jsonrpsee/blob/master/core/src/client/async_client/mod.rs#L458-L490). We could also forward them, but then we'd need to make sure that we stay up-to-date with these functions.
